### PR TITLE
Add Resource Detectors to frontend service

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

This will add resource detectors to the frontend service, to create richer data for the backend, e.g.:

Screenshot 2022-12-20 at 16 51 37

Similar could be done for other languages/services, I just wanted to get started with Node.JS where I know how to do it :-) 

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

### CODE REVIEW

**Review by julianocosta89 — APPROVED** [review]
@svrnm this is really cool!
Thanks for this PR.

We have something similar in Rust, but not with that many resourceDetectors.

**Comment by svrnm** [comment]
> @svrnm this is really cool! Thanks for this PR.

Happy to help & that you like it:-)
I will try to add some more eventually across the demo services :-)

> We have something similar in Rust, but not with that many resourceDetectors.

There are implementations of container resource detection & cloud resource detection across many languages, so they can be used as a template for rust, since the mechanics are always the same (e.g. for container.id you need to read it from the right files, and for the cloud ones there are APIs).

**Review by fatsheep9146 — APPROVED** [review]

**Comment by cartersocha** [comment]
@puckpuck / @julianocosta89 I feel like we should document the resource enrichment somewhere? Maybe a dedicated file to keep things simple or add to the attributes file

**Comment by cartersocha** [comment]
Thanks for the addition @svrnm! This looks great 🚀

File: src/frontend/package.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
     "@opentelemetry/exporter-trace-otlp-grpc": "0.33.0",
     "@opentelemetry/exporter-trace-otlp-http": "0.33.0",
     "@opentelemetry/instrumentation": "0.33.0",
+    "@opentelemetry/resource-detector-alibaba-cloud": "^0.27.3",
+    "@opentelemetry/resource-detector-aws": "^1.2.1",
+    "@opentelemetry/resource-detector-container": "^0.2.1",
+    "@opentelemetry/resource-detector-gcp": "^0.27.4",
```
Comment (mic-max) [inline src/frontend/package.json:25]: Consistent version pinning
```suggestion
    "@opentelemetry/resource-detector-alibaba-cloud": "0.27.3",
    "@opentelemetry/resource-detector-aws": "1.2.1",
    "@opentelemetry/resource-detector-container": "0.2.1",
    "@opentelemetry/resource-detector-gcp": "0.27.4",
```

**Review by mic-max — APPROVED** [review]

**Comment by puckpuck** [comment]
@cartersocha is something to discuss in the SIG meeting in the new year, but I think we can put Resource detection in the feature/metric matrix tables. We should also document how it is accomplished in each service doc, similar to how we show other instrumentation features.

**Review by puckpuck — APPROVED** [review]

**Pull request merged**